### PR TITLE
Radio

### PR DESCRIPTION
**Pull request by sidra2022**

### CODE REVIEW

File: packages/stencil/src/components/datacom-radio/datacom-radio.css [hunk 1]
```diff
@@ -49,124 +47,142 @@
...
+  border: 1px solid var(--dc-greyscale-05);
+  display: flex;
+  background-color: var(--dc-background-colour);
+  font-family: var(--dc-font-family);
+  font-weight: 500;
+  align-items: center;
 }
 
-.dc-radio-buttons.dc-radio-size-standard + label {
+/* for standar size button and */
```
Comment (AngeCast) [inline packages/stencil/src/components/datacom-radio/datacom-radio.css:139]: Misspelt the word "standard". Also you ended the sentence with an 'and'? Is this comment incomplete?

File: packages/stencil/src/components/datacom-radio/datacom-radio.css [hunk 2]
```diff
@@ -49,124 +47,142 @@
...
+.dc-radio-grouped.dc-radio-size-small .dc-radio-input + label,
+.dc-radio-buttons.dc-radio-size-small .dc-radio-input + label {
   height: 34px;
   font-size: 14px;
   padding-left: 12px;
   padding-right: 12px;
 }
-
-.dc-radio-image {
+/* image dispaly  */
```
Comment (AngeCast) [inline packages/stencil/src/components/datacom-radio/datacom-radio.css:155]: MIsspelt "display"

File: packages/stencil/src/components/datacom-radio/datacom-radio.tsx [hunk 3]
```diff
@@ -1,10 +1,11 @@
-import { Component, Host, h, Prop, Element, EventEmitter, Event, VNode, State, Method } from '@stencil/core';
+import { Component, h, Prop, Element, EventEmitter, Event, VNode, State, Method, Host } from '@stencil/core';
 import { getSvg } from '../../common/images/icon-provider';
 import { randomString } from '../../utils';
 import { FormControl } from '../form-control';
 
 export type RadioSize = 'standard' | 'small';
-export type RadioVariant = 'radios' | 'buttons';
+export type RadioVariant = 'radios' | 'buttons' | 'grouped';
```
Comment (AngeCast) [inline packages/stencil/src/components/datacom-radio/datacom-radio.tsx:7]: Should rather just use 'radio' and 'button'. No need for plurals.

Also, you don't need this grouped variant? We introduced all that state work so we didn't have to have an extra variant, right?

File: packages/stencil/src/components/datacom-radio/datacom-radio.tsx [hunk 4]
```diff
@@ -55,14 +58,21 @@ export class DatacomRadio implements FormControl {
   private inputElement: HTMLInputElement;
 
   handleChange = () => {
+    console.log('handlechange', this.checked);
```
Comment (AngeCast) [inline packages/stencil/src/components/datacom-radio/datacom-radio.tsx:61]: Shouldn't leave console logs in final code (unless in some dire debugging situations)

File: packages/stencil/src/components/datacom-radio/datacom-radio.tsx [hunk 5]
```diff
@@ -55,14 +58,21 @@ export class DatacomRadio implements FormControl {
...
+
     this.checked = true;
   };
 
-  /**
-   * Check if control is valid (for form submit)
-   *
-   * @returns Promise of validity
-   */
+  // * control use for radio button group   */
```
Comment (AngeCast) [inline packages/stencil/src/components/datacom-radio/datacom-radio.tsx:66]: Please use multi line comment syntax for consistency with other function comment formats

```
/**
 * Comment goes here.
 */
```

File: packages/stencil/src/components/datacom-radio/datacom-radio.tsx [hunk 6]
```diff
@@ -98,86 +145,60 @@ export class DatacomRadio implements FormControl {
...
       }
 
       const classes = {
+        'dc-radio-grouped': this.grouped,
         'dc-radio-disabled': this.disabled,
         'dc-radio-checked': this.checked,
         [`dc-radio-${variant}`]: true,
         [`dc-radio-size-${size}`]: true,
-        [this.type]: true,
+        // [this.type]: true,
```
Comment (AngeCast) [inline packages/stencil/src/components/datacom-radio/datacom-radio.tsx:161]: Do you need this code?

File: packages/stencil/src/components/datacom-radio/datacom-radio.tsx [hunk 7]
```diff
@@ -98,86 +145,60 @@ export class DatacomRadio implements FormControl {
...
+                  disabled={this.disabled}
+                  required={this.required}
+                  value={this.value}
+                  id={this.inputId}
+                  onChange={this.handleChange}
+                  tabIndex={0}
+                />
+                <label tabIndex={0} htmlFor={this.inputId} class="dc-radio-label">
+                  <span tabIndex={-1} class={`dc-radio-image-${this.imagePosition}`}>
+                    {image}
```
Comment (AngeCast) [inline packages/stencil/src/components/datacom-radio/datacom-radio.tsx:191]: Didn't you need to conditionally remove this image depending on if it's a 'radio' or 'button' variant?

File: packages/stencil/src/components/datacom-radio/useage.md [hunk 8]
```diff
@@ -7,5 +7,12 @@
 
   <datacom-radio label="Radio" id="radio1" value="choice3" name="choose" variant="buttons" size="small" disabled="true"> </datacom-radio>
```
Comment (AngeCast) [inline packages/stencil/src/components/datacom-radio/useage.md:8]: Could you rename this file to usage.md? 'Useage' is actually misspelt.

File: packages/stencil/src/index.html [hunk 9]
```diff
@@ -14,6 +14,15 @@
 <body>
   <datacom-menubar></datacom-menubar>
 
+<<<<<<< HEAD
```
Comment (AngeCast) [inline packages/stencil/src/index.html:17]: You've got merge conflicts in this file (evident by this HEAD line here). Please address.

**Review by AngeCast — COMMENTED** [review]
Got a bunch of comments. Please address when you can.

File: packages/stencil/src/components.d.ts [hunk 10]
```diff
@@ -328,6 +328,7 @@ export namespace Components {
         "name": string;
```
[the remark this answers is not in the document]
Comment (sidra2022) [reply packages/stencil/src/components.d.ts]: Fixed it.

File: packages/stencil/src/components/datacom-radio/datacom-radio-group.tsx [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+export class DatacomRadioGroup {
+  @Element() host: HTMLElement;
+  @Prop() child: boolean;
+  /*  for variant display layout  when wrap with group radio component  */
+  @Prop() horizontal = true;
+
+  private children: HTMLDatacomRadioElement[] = [];
+  /**
+   * Gets a list of all immediate children of this group element
+   *
```
[the remark this answers is not in the document]
Comment (sidra2022) [reply packages/stencil/src/components/datacom-radio/datacom-radio-group.tsx:18]: Fixed it.

File: packages/stencil/src/components/datacom-radio/datacom-radio.css [hunk 12]
```diff
@@ -1,40 +1,38 @@
 :host {
```
[the remark this answers is not in the document]
Comment (sidra2022) [reply packages/stencil/src/components/datacom-radio/datacom-radio.css:1]: Fixed it.

File: packages/stencil/src/components/datacom-radio/datacom-radio.css [hunk 13]
```diff
@@ -49,124 +47,142 @@
...
+/* For grouped */
+.dc-radio-grouped .dc-radio-input,
+.dc-radio-buttons .dc-radio-input {
   position: absolute;
   cursor: pointer;
-  flex-direction: row;
-  opacity: 0;
-  width: 100px;
-  height: 60px;
+  /* flex-direction: row; */
```
[the remark this answers is not in the document]
Comment (sidra2022) [reply packages/stencil/src/components/datacom-radio/datacom-radio.css]: Fixed it.

File: packages/stencil/src/components/datacom-radio/datacom-radio.css (hunk 1 above)
Comment (sidra2022) [reply packages/stencil/src/components/datacom-radio/datacom-radio.css]: Fixed it.

File: packages/stencil/src/components/datacom-radio/datacom-radio.tsx (hunk 5 above)
Comment (sidra2022) [reply packages/stencil/src/components/datacom-radio/datacom-radio.tsx]: Fixed it.

File: packages/stencil/src/components/datacom-radio/datacom-radio.tsx (hunk 6 above)
Comment (sidra2022) [reply packages/stencil/src/components/datacom-radio/datacom-radio.tsx]: Fixed it.

File: packages/stencil/src/components/datacom-radio/datacom-radio.tsx (hunk 7 above)
Comment (sidra2022) [reply packages/stencil/src/components/datacom-radio/datacom-radio.tsx]: Fixed it.

File: packages/stencil/src/components/datacom-radio/useage.md (hunk 8 above)
Comment (sidra2022) [reply packages/stencil/src/components/datacom-radio/useage.md:8]: Fixed it.

File: packages/stencil/src/index.html (hunk 9 above)
Comment (sidra2022) [reply packages/stencil/src/index.html]: Fixed it.

**Pull request merged**